### PR TITLE
DCS-1534 Sanitising search criteria before calling api

### PR DESCRIPTION
**Pull request by andrewrlee**

### CODE REVIEW

File: server/@types/welcome/index.d.ts [hunk 1]
```diff
@@ -96,9 +96,9 @@ declare module 'welcome' {
       description: string
     }
     PotentialMatchCriteria: {
-      firstName: string
-      lastName: string
-      dateOfBirth: string
+      firstName?: string
+      lastName?: string
+      dateOfBirth?: string
```
Comment (andrewrlee) [inline server/@types/welcome/index.d.ts:101]: These are now all optional

File: server/routes/bookedtoday/arrivals/reviewPerDetailsChangeDateOfBirthController.ts [hunk 2]
```diff
@@ -22,10 +23,7 @@ export default class ReviewPerDetailsChangeDateOfBirthController {
       }
 
       const { day, month, year } = req.body
-
-      State.newArrival.update(req, res, {
-        dateOfBirth: `${year.padStart(4, '0')}-${month.padStart(2, '0')}-${day.padStart(2, '0')}`,
```
Comment (andrewrlee) [inline server/routes/bookedtoday/arrivals/reviewPerDetailsChangeDateOfBirthController.ts:27]: we were doing this padding all over so moved to utils

File: server/routes/bookedtoday/arrivals/state.ts [hunk 3]
```diff
@@ -53,8 +53,7 @@ const SearchDetailsCodec: Codec<SearchDetails> = {
   write: (value: SearchDetails): Record<string, string> => ({ ...value }),
 
   read(record: Record<string, unknown>): SearchDetails {
-    assertHasStringValues(record, ['firstName', 'lastName', 'dateOfBirth'])
-    assertHasOptionalStringValues(record, ['prisonNumber', 'pncNumber'])
+    assertHasOptionalStringValues(record, ['firstName', 'lastName', 'dateOfBirth', 'prisonNumber', 'pncNumber'])
```
Comment (andrewrlee) [inline server/routes/bookedtoday/arrivals/state.ts:56]: all now optional

File: server/routes/bookedtoday/unexpectedArrivals/searchForExistingRecordsController.test.ts [hunk 4]
```diff
@@ -146,6 +146,18 @@ describe('Unexpected arrivals - search for existing records', () => {
...
 
+    it('can perform prison number search', () => {
+      return request(app)
+        .post('/manually-confirm-arrival/search-for-existing-record')
+        .send({ prisonNumber: 'A1234AA', day: '', month: '', year: '' })
+        .expect(302)
+        .expect(() => {
+          expect(expectedArrivalsService.getMatchingRecords).toHaveBeenCalledWith({
+            prisonNumber: 'A1234AA',
+          })
```
Comment (andrewrlee) [inline server/routes/bookedtoday/unexpectedArrivals/searchForExistingRecordsController.test.ts:157]: this previously blew up as would have passed a dob of '0000-00-00' to the backend

File: server/routes/bookedtoday/unexpectedArrivals/searchForExistingRecordsController.ts [hunk 5]
```diff
@@ -34,16 +34,7 @@ export default class SearchForExistingRecordsController {
...
-      const dateOfBirth = `${year.padStart(4, '0')}-${month.padStart(2, '0')}-${day.padStart(2, '0')}`
-
-      const searchData = {
-        firstName: convertToTitleCase(firstName),
-        lastName: convertToTitleCase(lastName),
-        dateOfBirth,
-        prisonNumber: prisonNumber || undefined,
-        pncNumber: pncNumber || undefined,
-      }
+      const searchData = sanitiseSearchCriteria(req.body)
```
Comment (andrewrlee) [inline server/routes/bookedtoday/unexpectedArrivals/searchForExistingRecordsController.ts:37]: extracted out all the formatting and build up of data

**Review by michalnawrockidj — APPROVED** [review]
LGTM

**Pull request merged**